airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/http_selector.py

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, List, Mapping
+
+import requests
+from airbyte_cdk.sources.declarative.types import Record
+
+
+class HttpSelector(ABC):
+    @abstractmethod
+    def select_records(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
```

Comment:
> @brianjlai open question: would it make more sense for the extractor to have access to the state object instead of passing in the stream_state, stream_slice, and next_page_token directly?
It seems like a more generic pattern we could apply to a few more classes

picking back up this discussion point got buried since the file was outdated. Its a bit verbose but I actually like explicitly passing these 3 fields into extract/filter/transform. We always have to receive those parameters since they adhere to the existing CDK interface for `parse_response()` and I think its clearer to know what parameters we have access to. We could pass in `self._state`, but it feels more ambiguous as to how we should access state and use it.